From a57366d6d050945bd5d18e4fd4d9f2f7cf8aec70 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:28:10 -0400
Subject: [tools] perf/header: Fix bogus group name

Message-id: <1408462094-14194-322-git-send-email-jolsa@redhat.com>
Patchwork-id: 88262
O-Subject: [PATCH RHEL7.1 BZ1131394 321/325] perf header: Fix bogus group name
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 210e812f036736aeda097d9a6ef84b1f2b334bae
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Mon Nov 18 11:20:43 2013 +0900

description
===========
When processing event group descriptor in perf file header, we reuse an
allocated group name but forgot to prevent it from freeing.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/header.c b/tools/perf/util/header.c
index 369c036..559c516 100644
--- a/tools/perf/util/header.c
+++ b/tools/perf/util/header.c
@@ -2078,8 +2078,10 @@ static int process_group_desc(struct perf_file_section *section __maybe_unused,
   if (evsel->idx == (int) desc[i].leader_idx) {
    evsel->leader = evsel;
    /* {anon_group} is a dummy name */
-   if (strcmp(desc[i].name, "{anon_group}"))
+   if (strcmp(desc[i].name, "{anon_group}")) {
     evsel->group_name = desc[i].name;
+    desc[i].name = NULL;
+   }
    evsel->nr_members = desc[i].nr_members;
 
    if (i >= nr_groups || nr > 0) {
-- 
1.7.1